File: ethereum/compile.js
const path = require('path')
const fs = require('fs-extra')
const exec = require('child_process').exec

const buildPath = path.resolve(__dirname, 'build')
fs.removeSync(buildPath) // delete folder and its content

const contractsPath = path.resolve(__dirname, 'contracts', 'Hash.vy')

// recreate build folder
fs.ensureDirSync(buildPath)

// compile contract and save ABI output
exec(`vyper -f abi ${contractsPath} > ${path.resolve(buildPath, 'hash.json')}`)

// save bytecode
exec(`vyper -f bytecode ${contractsPath} > ${path.resolve(buildPath, 'hashBytecode')}`)

File: ethereum/compile.py
from vyper import compiler
import json


def compile(source_code):
    out = compiler.compile_code(
        source_code,
        ['abi', 'bytecode']
    )
    return out


with open('./contract/Hash.vy') as f:
    contract_code = f.read()
    contract = compile(contract_code)
    f.close()

with open('../client/src/ethereum/hash.json', 'w', encoding='utf-8') as f:
    json.dump(contract, f, ensure_ascii=False, indent=2)
    f.close()
